From dfc0d40fffe70821461564911699d19af7e81001 Mon Sep 17 00:00:00 2001
From: Satish Kumar Nagireddy <satish.nagireddy.nagireddy@xilinx.com>
Date: Tue, 6 Mar 2018 17:49:42 -0800
Subject: [PATCH 0322/1852] v4l: xilinx: dma: Fix bytes per line calculation

commit f4886d2dfaa0344ec4758f5e56ef337213039ce2 from
https://github.com/Xilinx/linux-xlnx.git

In current implementation there is a bug where min_bpl value
is not satisfying dma alignment. This patch will fix the issue
by aligning the value properly.

Signed-off-by: Satish Kumar Nagireddy <satish.nagireddy.nagireddy@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-dma.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/media/platform/xilinx/xilinx-dma.c b/drivers/media/platform/xilinx/xilinx-dma.c
index bbeaad9a6eed..59fca8dc4d26 100644
--- a/drivers/media/platform/xilinx/xilinx-dma.c
+++ b/drivers/media/platform/xilinx/xilinx-dma.c
@@ -835,6 +835,7 @@ __xvip_dma_try_format(struct xvip_dma *dma,
 				min_bpl = (plane_width * info->bpl_factor *
 					   padding_factor_nume * bpl_nume) /
 					   (padding_factor_deno * bpl_deno);
+				min_bpl = roundup(min_bpl, dma->align);
 				bpl = rounddown(plane_fmt[i].bytesperline,
 						dma->align);
 				plane_fmt[i].bytesperline =
-- 
2.31.1

